navigation-fragments/src/main/java/dev/hotwire/navigation/navigator/NavigatorHost.kt

```diff
@@ -63,6 +71,22 @@ open class NavigatorHost : NavHostFragment(), FragmentOnAttachListener {
         }
     }
 
+    @VisibleForTesting(otherwise = PROTECTED)
+    fun ensureDeeplinkStartLocationValid() {
+        val extrasBundle = activity.intent.extras?.getBundle(DEEPLINK_EXTRAS_KEY) ?: return
+        val startLocation = extrasBundle.getString(LOCATION_KEY) ?: return
+
+        val deepLinkStartUri = startLocation.toUriOrNull()
+        val configStartUri = configuration.startLocation.toUriOrNull()
+
+        if (deepLinkStartUri?.host != configStartUri?.host) {
+            extrasBundle.putString(LOCATION_KEY, configuration.startLocation)
+            activity.intent.putExtra(DEEPLINK_EXTRAS_KEY, extrasBundle)
+        }
+    }
+
+    private fun String.toUriOrNull() = runCatching { toUri() }.getOrNull()
```

Comment:
Is this exception catching necessary? From what I'm aware, this can only throw a `NullPointerException` in the Java implementation, which is guarded by the compiler in the Kotlin extension.